src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -236,5 +238,49 @@ public void Build(ICapabilityStatementBuilder builder)
                 builder.AddRestInteraction(SystemRestfulInteraction.Batch);
             }
         }
+
+        public async Task<ResourceWrapper> ReplaceAsync(ResourceWrapper resourceWrapper, WeakETag weakETag, CancellationToken cancellationToken)
```

Comment:
If in the future we can send multiple documents, should this take an IEnumerable<ResourceWrapper>?

Reply:
My thought was that when we add support for updating multiple documents, we should add a separate API called `BatchReplaceAsync`? This `ReplaceAsync` method could still be used for other scenarios where we are looking to replace just a single document. I was approaching this more from a generic api for a datastore. I don't have a strong preference and am open to updating this method to take in an IEnumerable